src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -566,6 +566,22 @@ CreateCombineQueryForRouterPlan(DistributedPlan *distPlan)
 	combineQuery->querySource = QSRC_ORIGINAL;
 	combineQuery->canSetTag = true;
 	combineQuery->rtable = list_make1(rangeTableEntry);
+
+#if PG_VERSION_NUM >= PG_VERSION_16
+
+	/*
+	 * This part of the code is more of a sanity check for readability,
+	 * it doesn't really do anything.
+	 * We know that Only relation RTEs and subquery RTEs that were once relation
```

Comment:
I copy-pasted this comment from PG16. So, normally it would be just for relation RTEs, but PG16 made these changes to views in which they convert them to subquery RTEs. Given that we need to check permissions for views as well, they added that in the list. So, it becomes "relation RTEs and subquery RTEs that were once relation RTEs i.e. views"